src/main/java/reactor/core/publisher/FluxPeekFuseable.java

```diff
@@ -258,7 +263,7 @@ public T poll() {
 					throw Exceptions.propagate(Operators.onOperatorError(s, e, v));
 				}
 			}
-			if (v == null && sourceMode == SYNC) {
+			if (v == null && (d || sourceMode == SYNC)) {
```

Comment:
This avoids a volatile read for sync use but doubles the code size, reduces inline chances, etc.
```java
		    if (sourceMode == SYNC) {
	            T v = s.poll();
	            if (v != null && parent.onNextCall() != null) {
	                try {
	                    parent.onNextCall().accept(v);
	                }
	                catch (Throwable e) {
	                    throw Exceptions.propagate(Operators.onOperatorError(s, e, v));
	                }
	            }
	            if (v == null) {
	                Runnable call = parent.onCompleteCall();
	                if (call != null) {
	                    call.run();
	                }
	                call = parent.onAfterTerminateCall();
	                if (call != null) {
	                    call.run();
	                }
	            }
	            return v;
		    }
            boolean d = done;
            T v = s.poll();
            if (v != null && parent.onNextCall() != null) {
                try {
                    parent.onNextCall().accept(v);
                }
                catch (Throwable e) {
                    throw Exceptions.propagate(Operators.onOperatorError(s, e, v));
                }
            }
            if (v == null && d) {
                Runnable call = parent.onCompleteCall();
                if (call != null) {
                    call.run();
                }
                call = parent.onAfterTerminateCall();
                if (call != null) {
                    call.run();
                }
            }
            return v;
```